src/Core/src/Platform/Windows/SearchBarExtensions.cs

```diff
@@ -94,15 +94,23 @@ public static void UpdateVerticalTextAlignment(this AutoSuggestBox platformContr
 
 		public static void UpdateMaxLength(this AutoSuggestBox platformControl, ISearchBar searchBar)
 		{
+			var maxLength = searchBar.MaxLength;
+
+			if (maxLength == -1)
+				maxLength = int.MaxValue;
+
+			if (maxLength == 0)
+				MauiAutoSuggestBox.SetIsReadOnly(platformControl, true);
+
 			var currentControlText = platformControl.Text;
 
-			if (currentControlText.Length > searchBar.MaxLength)
-				platformControl.Text = currentControlText.Substring(0, searchBar.MaxLength);
+			if (currentControlText.Length > maxLength)
+				platformControl.Text = currentControlText.Substring(0, maxLength);
 		}
-		
+
 		public static void UpdateIsReadOnly(this AutoSuggestBox platformControl, ISearchBar searchBar)
 		{
-			platformControl.IsEnabled = searchBar.IsReadOnly;
+			MauiAutoSuggestBox.SetIsReadOnly(platformControl, searchBar.IsReadOnly);
```

Comment:
If I set IsReadonly=False and MaxLength to 0 and then it gets disabled, can I re-enable by just setting IsReadOnly=True? Should IsReadonly reall be a combination of `searchBar.IsReadOnly && MaxLength > 0`.

Same with the TextBox below.

Reply:
good point, will check on those and add tests!